src/coreclr/jit/emitarm64.cpp

```diff
@@ -7940,16 +7935,19 @@ void emitter::emitIns_R_I(instruction     ins,
 
         case INS_sve_mov:
         case INS_sve_dup:
-            optionalShift = true;
-            hasShift      = (sopt == INS_SCALABLE_OPTS_SHIFT);
-
             assert(insOptsScalableStandard(opt));
-            // Size specifier must be able to fit left-shifted immediate
-            assert(!hasShift || insOptsScalableAtLeastHalf(opt));
-            assert(insScalableOptsNone(sopt) || hasShift);         // h
             assert(isVectorRegister(reg));                         // ddddd
-            assert(isValidSimm8(imm));                             // iiiiiiii
             assert(isValidVectorElemsize(optGetSveElemsize(opt))); // xx
+
+            if (!isValidSimm8(imm))
+            {
+                // Size specifier must be able to fit a left-shifted immediate
+                assert(isValidSimm8_MultipleOf256(imm)); // iiiiiiii
+                assert(insOptsScalableAtLeastHalf(opt));
+                hasShift = true;
```

Comment:
So for `B` element, essentially the `sh` bit is ignored, which makes sense, but it is not clearly stated in the [docs](https://docsmirror.github.io/A64/2023-06/dup_z_i.html).

Reply:
Yeah, I don't see it clearly stated either, though I do see this line in the pseudocode description:
`if size:sh == '001' then UNDEFINED;`
Which means the behavior is undefined if the size specifier is `B`, but the `sh` bit is 1. I think these asserts reflect that.